### PR TITLE
New package: startup-1.13.3

### PR DESCRIPTION
**Pull request by CameronNemo**

### CODE REVIEW

**Comment by travankor** [comment]
Is this just a rebranded version of upstart ?

**Comment by CameronNemo** [comment]
Pretty much. Upstart is trademarked by Canonical, and the last released version doesn't compile on glibc or musl libc. I made it build, pulled in Chrome OS and Ubuntu patches, and fixed some bugs myself. You can see the NEWS file, ChangeLog, or git history for a summary.

File: srcpkgs/startup/template [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Template file for 'startup'
+pkgname=startup
+version=1.13.3
+revision=1
+build_style=gnu-configure
+configure_args="--disable-sysvcompat --disable-dconf-bridge --disable-cgroups"
```
[the remark this answers is not in the document]
Comment (CameronNemo) [reply srcpkgs/startup/template]: Simply that I am not using or testing it, and it adds a dependency on cgmanager. cgmanager can sometimes configure cgroups in a way that doesn't interact well with container runtimes (see #5402).

File: srcpkgs/startup/template [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+		vmove usr/share/applications/startup-monitor.desktop
+		vmove usr/share/man/man8/startup-monitor.8
+	}
+}
+
+startup-udev-bridge_package() {
+	short_desc+=" - udev event bridge"
+	pkg_install() {
+		vmove usr/libexec/startup/bridge/udev
+		vmove etc/init/bridge-udev.conf
```
Comment (maxice8) [inline srcpkgs/startup/template]: should problably have conf_files

File: srcpkgs/startup/template [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+}
+
+libupstart-devel_package() {
+	depends="libupstart>=${version}_${revision}"
+	short_desc+=" - library (development files)"
+	pkg_install() {
+		vmove usr/include
+		vmove usr/lib/pkgconfig
+		vmove "usr/lib/*.a"
+		vmove "usr/lib/*.la"
```
Comment (maxice8) [inline srcpkgs/startup/template]: do we need to keep the libtool archives ?

File: srcpkgs/startup/template (hunk 1 above)
[the remark this answers is not in the document]
Comment (maxice8) [reply srcpkgs/startup/template]: I see, fair enough.

File: srcpkgs/startup/template [hunk 4]
```diff
@@ -0,0 +1,66 @@
+# Template file for 'startup'
+pkgname=startup
+version=1.13.3
+revision=1
+build_style=gnu-configure
+configure_args="--disable-sysvcompat --disable-dconf-bridge --disable-cgroups"
+hostmakedepends="autoconf automake libtool pkg-config gettext-devel xz nih-devel"
```
Comment (maxice8) [inline srcpkgs/startup/template]: xz is part of base-devel

File: srcpkgs/startup/template [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		vmove usr/bin/initctl2dot
+		vmove usr/bin/startup-monitor
+		vmove usr/share/icons
+		vmove usr/share/startup/icons
+		vmove usr/share/applications/startup-monitor.desktop
+		vmove usr/share/man/man8/startup-monitor.8
+	}
+}
+
+startup-udev-bridge_package() {
```
Comment (maxice8) [inline srcpkgs/startup/template]: why have udev-bridge separated from other bridges ?

File: srcpkgs/startup/template [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+short_desc="Event driven task and service manager"
+maintainer="Cameron Nemo <camerontnorman@gmail.com>"
+license="GPL-2.0-only"
+homepage="https://gitlab.com/chinstrap/startup/startup"
+changelog="${homepage}/raw/master/NEWS"
+distfiles="${homepage}/-/archive/${version}/startup-${version}.tar.gz"
+checksum=b63de554d039db6cc920a50a4d38f83529c362ef725dece557247524a84d78ae
+
+make_dirs="/var/log/upstart 0755 root root"
+
```
Comment (maxice8) [inline srcpkgs/startup/template]: shouldn't we have

```
/etc/init/bridge-dbus.conf
/etc/init/bridge-file.conf
/etc/init/bridge-socket.conf
```

in conf_files ?

File: srcpkgs/startup/template (hunk 3 above)
Comment (CameronNemo) [reply srcpkgs/startup/template]: You tell me. Should I rm it in post_install()? I am not well versed in the subject, a little research yielded a probably. Do they hurt if they are present?

File: srcpkgs/startup/template (hunk 5 above)
Comment (CameronNemo) [reply srcpkgs/startup/template]: libudev dependency. All the other bridges introduce no new dependencies. If the dconf bridge is enabled later I intend to make it a subpackage.

File: srcpkgs/startup/template (hunk 4 above)
Comment (CameronNemo) [reply srcpkgs/startup/template]: I feel like you have told me this before :)

File: srcpkgs/startup/template (hunk 3 above)
Comment (maxice8) [reply srcpkgs/startup/template]: it should normally be removed unless keep_libtool_archives are present.

File: srcpkgs/startup/template (hunk 5 above)
Comment (maxice8) [reply srcpkgs/startup/template]: libudev is already part of base-system.

**Comment by maxice8** [comment]
On Mon, Dec 31, 2018 at 02:26:29PM -0800, Cameron Nemo wrote:
> CameronNemo commented on this pull request.
> 
> 
> 
> > +}
> +
> +startup-tools_package() {
> +	depends="${sourcepkg}>=${version}_${revision} python3-dbus python3-gobject"
> +	short_desc+=" - monitoring and analysis tools"
> +	pkg_install() {
> +		vmove usr/bin/initctl2dot
> +		vmove usr/bin/startup-monitor
> +		vmove usr/share/icons
> +		vmove usr/share/startup/icons
> +		vmove usr/share/applications/startup-monitor.desktop
> +		vmove usr/share/man/man8/startup-monitor.8
> +	}
> +}
> +
> +startup-udev-bridge_package() {
> 
> Indeed, but a busybox based rootfs may not have it.
> 
> -- 
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/void-linux/void-packages/pull/6375#discussion_r244621476

We have a busybox rootfs ?

File: srcpkgs/startup/template [hunk 7]
```diff
@@ -0,0 +1,75 @@
...
+libupstart_package() {
+	short_desc+=" - library"
+	pkg_install() {
+		vmove "usr/lib/libupstart.so.*"
+	}
+}
+
+libupstart-devel_package() {
+	depends="libupstart>=${version}_${revision}"
+	short_desc+=" - library (development files)"
```
Comment (maxice8) [inline srcpkgs/startup/template]: can be just `development files`

Comment (CameronNemo) [reply srcpkgs/startup/template]: done.

**Pull request merged**